File: DSA/Aditi-Sharma/Day13/day13_question2.java
import java.util.*;
class ispalindrome
{
static int rev(int n, int temp)
{
	if (n == 0)
    {
        return temp;
    }
	temp = (temp * 10) + (n % 10);
	return rev(n / 10, temp);
}

public static void main (String[] args) 
{
    Scanner sc=new Scanner(System.in);
	int n = sc.nextInt();
	int temp = rev(n, 0);
	if (temp == n)
		System.out.println("yes");
	else
		System.out.println("no" );
}
}

File: DSA/Pragya_Agarwal/Day13_Question2.py
def palindrome(num,temp=None,rev=None):
    if rev is None:
        rev = 0
    if temp == 0:
        return num == rev
    if temp is None:
        temp = num
    digit = temp % 10
    rev = rev * 10 + digit
    return palindrome(num, temp // 10, rev)

num = 121
if palindrome(num):
    print(num,'is palindrome')
else:
    print(num,'is not a palindrome')
